hydroflow_plus/tests/compile_fail.rs

```diff
@@ -1,3 +1,5 @@
+#![cfg(nightly)]
```

Comment:
Oh does trybuild not work on nightly??

Reply:
So interesting setup, for nightly dev we have `.cargo/config.toml` set up to set nightly flags namely `-Zproc-macro-backtrace`, but `-Z` unstable options do not work on nightly so cargo immediately exits. We can override this on stable by setting `RUSTFLAGS=""`. When trybuild runs it tries to run with the `.cargo/config.toml` and compilation fails. There is a way to work around it, but in general the error messages will be different between nightly and stable, so we should just stick with one to snapshot test, and I went with nightly

Reply:
Could we just... not put `-Zproc-macro-backtrace`? I feel like that was important when we had more bugs in our macros but now they're mostly ironed out.

Reply:
Yeah that would work too....